ui_testing/testcases/header_tests/test08_usermanagement.py

```diff
@@ -238,4 +227,289 @@ def test010_update_user_role_with_save_cancel_btn(self, save):
+        self.header_page.set_user_email((self.header_page.generate_random_email()))
+        new_email = self.header_page.get_user_email()
+        if 'save_btn' == save:
+            self.header_page.save(save_btn='user_management:save_btn')
+        else:
+            self.header_page.cancel(force=True)
+
+            self.base_selenium.get(
+                url=user_url, sleep=self.base_selenium.TIME_MEDIUM)
+
+        user_email = self.header_page.get_user_email()
+        if 'save_btn' == save:
+            self.base_selenium.LOGGER.info(
+                    ' + Assert {} (new_email) == {} (user_email)'.format(new_email, user_email))
+            self.assertEqual(new_email, user_email)
+        else:
+            self.base_selenium.LOGGER.info(
+                    ' + Assert {} (current_email) == {} (user_email)'.format(current_email, user_email))
+            self.assertEqual(current_email, user_email)
+
+    def test011_validation_user_name_email_fields(self):
+        """
+        Header: User management: Make sure when the user update name & email then press on save button,
+        red border display and he can't save
+        LIMS-6121
+        :return:
+            """
+        # from the create mode it will redirect me to the active table
+        self.base_selenium.click(element='header:user_management_button')
+        self.header_page.get_random_user()
+        self.header_page.clear_user_name()
+        self.header_page.clear_user_email()
+        self.header_page.save(save_btn='user_management:save_btn')
+        self.base_selenium.LOGGER.info('Waiting for error message')
+        validation_result = self.base_selenium.wait_element(element='general:oh_snap_msg')
+        self.base_selenium.LOGGER.info('Assert error msg')
+        self.assertEqual(validation_result, True)
+
+    def test012_delete_user_used_in_other_entity(self):
+        """
+        User management: Make sure that you can't delete any user record If this record used in other entity
+        LIMS-6407
+        :return:
+        """
+        self.base_selenium.click(element='header:user_management_button')
+        # create new user with random data
+        user_random_name = self.generate_random_string()
+        user_random_email = self.base_page.generate_random_email()
+        self.header_page.create_new_user(user_name=user_random_name,
+                                         user_email=user_random_email,
+                                         user_role='Admin', user_password='1',
+                                         user_confirm_password='1')
+
+        self.base_selenium.LOGGER.info(
+            'search to make sure that the user created '.format(user_random_name))
+        created_user = self.header_page.search(user_random_name)[0]
+        user_data = self.base_selenium.get_row_cells_dict_related_to_header(row=created_user)
+
+        self.header_page.click_on_header_button()
+        self.base_selenium.click(element='header:logout')
```

Comment:
API + Can be implemented in a separate Class in the same module to overwrite the login in the setUp